python/lsst/ts/hvac/xml/hvac_mqtt_to_SAL_XML.py

```diff
@@ -19,153 +19,495 @@
+    """Parse the string value of the limits column by comparing it to known
+    regular expressions and extracting the minimum and maximum values.
+
+    Parameters
+    ----------
+    limits_string: `str`
+        The string containing the limits to parse.
+
+    Returns
+    -------
+    lower_limit: `float`
+        The lower limit
+    upper_limit: `float`
+        The upper limit
+
+    Raises
+    ------
+    ValueError
+        In case an unknown string pattern is found in the limits column.
+
+    """
+    lower_limit = -99.9
+    upper_limit = 99.9
+    matched = False
+    for pattern in [
+        r"^(-*\d+)/(-*\d+)$",
+        r"^(-*\d+) a (-*\d+)$",
+        r"^(-*\d+)% a (-*\d+)%$",
+    ]:
+        match = re.match(pattern, limits_string)
+        if match:
+            lower_limit = float(match.group(1))
+            upper_limit = float(match.group(2))
+            matched = True
+    if not matched:
+        if re.match(r"^\d$", limits_string):
+            lower_limit = 0
+            upper_limit = 100
+            matched = True
+    if not matched:
+        raise ValueError(f"Couldn't match limits_string {limits_string}")
+
+    return lower_limit, upper_limit
+
+
+def get_items_for_topic(topic):
+    """Convenience method to get all items for an generic topic.
+
+    Parameters
+    ----------
+    topic: `str`
+        The generic topic to get the XML items for.
+
+    Returns
+    -------
+    items: `dict`
+        A dict containing all items for the topic.
+
+    Notes
+    -----
+    The structure of the dictionary is exactly the same as for the items in
+    `hvac_topics`.
+
+    """
+    global hvac_topics
+    if not bool(hvac_topics):
+        collect_topics_and_items()
+    items = {}
+    for hvac_topic in hvac_topics:
+        tpc, item = extract_topic_and_item(hvac_topic)
+        if tpc == topic:
+            items[item] = hvac_topics[hvac_topic]
+    return items
+
+
+def _generic_collect_topics_and_items(row, topics, items):
+    """Collect XML topics and items from a row read from the CSV file produced
+    from the Excel file with the Rubin Observatory HVAC system information
+    sent by DATControl.
+
+    The input CSV file can be found in the python/data/input directory and was
+    produced by collecting the data in all tabs in the Excel file into one tab
+    and exporting that tab to CSV.
+
+    Parameters
+    ----------
+    row: `list`
+        A row read from the CSV file.
+    topics: `dict`
+        The dictionary to add the XML topic and item, found in the row, to.
+    items: enum.Enum
+        The Enum that is used to translate the HVAC topics and items to XML
+        topics and items.
+
+
+    Raises
+    -------
+    ValueError
+        In case a HVAC topic or item is not present in the Telemetry, Item or
+        Command Enums. This error will only be raised during development of the
+        code and serves to ensure that all HVAC topics and items are present in
+        the Enums.
```

Comment:
Nice -- especially if this check is done when starting the CSC (thus detecting an outdated IDL file).

Reply:
Thanks :)